qemu/tests/win_irq_check.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_test
+
+
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU windows guest vitio device irq check test
+
+    1) Start guest with virtio device.
+    2) Make sure driver verifier enabled in guest.
+    3) Get irq info in guest.
+    4) Check the value of irq number.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def irq_check(session, device_name):
+        """
+        Return if virtio device's irq number is positive or not
+        :param session: use for sending cmd
+        :param device_name: virtio device's name
+        """
+        msinfo_cmd = params["msinfo_cmd"]
+        if session.cmd_status(msinfo_cmd, timeout=360):
+            test.error("Run msinfo32 failed.")
+        status, irq_dev_info = session.cmd_status_output(params["irq_cmd"]
+                                                         % device_name)
+        if status:
+            test.fail("Can't get %s's irq info." % device_name)
+        irq_value = re.split('\s+', irq_dev_info)[1]
+        logging.info("irq number is %s" % irq_value)
+
+        if int(irq_value) > 0:
+            return True
+        return False
+
+    driver = params["driver_name"]
+    device_name = params["device_name"]
+    timeout = int(params.get("login_timeout", 360))
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    error_context.context("Boot guest with %s device" % driver, logging.info)
```

Comment:
I think "Boot guest with xxx device " is more suitable than "Boot guest with xxx driver"

Reply:
I do see you define device_name in the cfg file,then why not use the device name directly?

Reply:
I define device_name for the following code " irq_num = irq_check(session, device_name)"
And "Boot guest with balloon/viorng device" is better than "Boot guest with 'VirtIO Balloon Driver'/'VirtIO RNG Device' device".